src/Microsoft.Data/ArrowStringColumn.cs

```diff
@@ -0,0 +1,348 @@
+                while (ret.Count < length)
+                {
+                    ret.Add((string)GetValue(startIndex++));
+                }
+                return ret;
+            }
+        }
+
+        public override Field Field => new Field(Name, StringType.Default, NullCount != 0);
+
+        public override int MaxRecordBatchLength(long startIndex)
+        {
+            if (Length == 0)
+                return 0;
+            int offsetsBufferIndex = GetBufferIndexContainingRowIndex(ref startIndex);
+            Debug.Assert(startIndex <= Int32.MaxValue);
+            return _offsetsBuffers[offsetsBufferIndex].Length - (int)startIndex;
+        }
+
+        private int GetNullCount(long startIndex, int numberOfRows)
+        {
+            int nullCount = 0;
+            for (long i = startIndex; i < numberOfRows; i++)
+            {
+                if (!IsValid(i))
+                    nullCount++;
+            }
+            return nullCount;
+        }
+
+
+        public override Apache.Arrow.Array AsArrowArray(long startIndex, int numberOfRows)
+        {
+            int offsetsBufferIndex = GetBufferIndexContainingRowIndex(ref startIndex);
+            ArrowBuffer dataBuffer = numberOfRows == 0 ? ArrowBuffer.Empty : new ArrowBuffer(_dataBuffers[offsetsBufferIndex].ReadOnlyMemory);
+            ArrowBuffer offsetsBuffer = numberOfRows == 0 ? ArrowBuffer.Empty : new ArrowBuffer(_offsetsBuffers[offsetsBufferIndex].ReadOnlyMemory);
+            ArrowBuffer nullBuffer = numberOfRows == 0 ? ArrowBuffer.Empty : new ArrowBuffer(_nullBitMapBuffers[offsetsBufferIndex].ReadOnlyMemory);
+            int nullCount = GetNullCount(startIndex, numberOfRows);
+            return new StringArray(numberOfRows, offsetsBuffer, dataBuffer, nullBuffer, nullCount, (int)startIndex);
+        }
+
+        public override BaseColumn Sort(bool ascending = true) => throw new NotSupportedException();
+
+        public override BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapIndices = false, long numberOfNullsToAppend = 0)
+        {
+            ArrowStringColumn clone;
+            if (!(mapIndices is null))
+            {
+                if (mapIndices.DataType != typeof(long))
+                    throw new ArgumentException(Strings.MismatchedValueType + " PrimitiveColumn<long>", nameof(mapIndices));
+                if (mapIndices.Length > Length)
+                    throw new ArgumentException(Strings.MapIndicesExceedsColumnLenth, nameof(mapIndices));
+                clone = Clone(mapIndices as PrimitiveColumn<long>, invertMapIndices);
+            }
+            else
+            {
+                clone = Clone();
+            }
+            for (long i = 0; i < numberOfNullsToAppend; i++)
+            {
+                clone.Append(null);
+            }
+            return clone;
+        }
+
+        private ArrowStringColumn Clone(PrimitiveColumn<long> mapIndices = null, bool invertMapIndex = false)
```

Comment:
Do you really need default values on this private method? Isn't it only called in 1 place?